File: src/main/java/com/github/trcdeveloppers/clayium/common/blocks/ores/BlockLargeDenseClayOre.java
package com.github.trcdeveloppers.clayium.common.blocks.ores;

import com.github.trcdeveloppers.clayium.common.annotation.CBlock;
import com.github.trcdeveloppers.clayium.common.items.ItemClayShovel;
import net.minecraft.block.Block;
import net.minecraft.block.SoundType;
import net.minecraft.block.material.Material;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;

import javax.annotation.ParametersAreNonnullByDefault;

import static com.github.trcdeveloppers.clayium.common.creativetab.ClayiumCreativeTab.CLAYIUM;

@CBlock(registryName = "large_dense_clay_ore")
public class BlockLargeDenseClayOre extends Block {
    public BlockLargeDenseClayOre(Material material) {
        super(material);
        this.setCreativeTab(CLAYIUM);
        this.setLightLevel(0f);
        this.setHarvestLevel("pickaxe", 1);
        this.setResistance(5f);
        this.setHardness(3f);
        this.setSoundType(SoundType.STONE);
    }

    @SuppressWarnings("unused")
    public BlockLargeDenseClayOre() {
        this(Material.ROCK);
    }

    @Override
    @ParametersAreNonnullByDefault
    public boolean canHarvestBlock(IBlockAccess world, BlockPos pos, EntityPlayer player) {
        if (player.getHeldItemMainhand().getItem() instanceof ItemClayShovel) return true;
        return super.canHarvestBlock(world, pos, player);
    }
}

File: src/main/kotlin/com/github/trcdevelopers/clayium/common/blocks/ores/BlockLargeDenseClayOre.kt
package com.github.trcdevelopers.clayium.common.blocks.ores

import com.github.trcdevelopers.clayium.common.Clayium
import com.github.trcdevelopers.clayium.common.annotation.CBlock
import com.github.trcdevelopers.clayium.common.items.ItemClayShovel
import net.minecraft.block.Block
import net.minecraft.block.SoundType
import net.minecraft.block.material.Material
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockAccess
import javax.annotation.ParametersAreNonnullByDefault

@CBlock(registryName = "large_dense_clay_ore")
class BlockLargeDenseClayOre(material: Material) : Block(material) {
    init {
        creativeTab = Clayium.CreativeTab
        soundType = SoundType.STONE
        setLightLevel(0f)
        setHarvestLevel("pickaxe", 1)
        setResistance(5f)
        setHardness(3f)
    }

    @Suppress("unused")
    constructor() : this(Material.ROCK)

    @ParametersAreNonnullByDefault
    override fun canHarvestBlock(world: IBlockAccess, pos: BlockPos, player: EntityPlayer): Boolean {
        return if (player.heldItemMainhand.item is ItemClayShovel) true else super.canHarvestBlock(world, pos, player)
    }
}
